src/lib/OpenEXR/ImfCompressor.cpp

```diff
@@ -52,54 +52,30 @@ Compressor::uncompressTile (
 bool
 isValidCompression (Compression c)
 {
-    switch (c)
-    {
-        case NO_COMPRESSION:
-        case RLE_COMPRESSION:
-        case ZIPS_COMPRESSION:
-        case ZIP_COMPRESSION:
-        case PIZ_COMPRESSION:
-        case PXR24_COMPRESSION:
-        case B44_COMPRESSION:
-        case B44A_COMPRESSION:
-        case DWAA_COMPRESSION:
-        case DWAB_COMPRESSION: return true;
-
-        default: return false;
-    }
+    return isValidCompressionId (c);
 }
 
 bool
 isLossyCompression (Compression c)
 {
-    switch (c)
-    {
-        case B44_COMPRESSION:
-        case B44A_COMPRESSION:
-        case DWAA_COMPRESSION:
-        case DWAB_COMPRESSION: return true;
-        default: return false;
-    }
+    return isLossyCompressionId (c);
```

Comment:
I didn't want to change `Compressor`'s API while keeping the codec API implementation in `ImfCompression.cpp`, so that developers would be able to register their new codec in a single location. That was the spirit of this work.
Are you suggesting we get rid of `isLossyCompression` and other methods that could be replaced by a direct call to the new API points ?

Reply:
I think I'm still misunderstanding something. There are now two functions which are part of the public API: `isLossyCompression` and `isLossyCompressionId`. `isLossyCompression` now just calls `isLossyCompressionId` but both are available to developers because both are declared in installed header files (albeit separate ones), and it's not clear which to call.

How about moving all the new code from `ImfCompression.cpp` into `ImfCompressor.cpp`, then you can implement `isLossyCompression` etc in ImfCompressor.cpp using the `IdToDesc` and you don't need any of the new duplicated names?